src/foundry/common/documents/roll-table.mjs.d.ts

```diff
@@ -0,0 +1,152 @@
+// FOUNDRY_VERSION: 10.291
+
+import Document, { DocumentMetadata } from "../abstract/document.mjs";
+import * as documents from "./module.mjs";
+import * as fields from "../data/fields.mjs";
+
+declare global {
+  type RollTableData = BaseRollTable.Properties;
+}
+
+/**
+ * The Document definition for a RollTable.
+ * Defines the DataSchema and common behaviors for a RollTable which are shared between both client and server.
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface BaseRollTable extends BaseRollTable.Properties {}
+declare class BaseRollTable extends Document<BaseRollTable.SchemaField, BaseRollTable.Metadata> {
+  /**
+   * @param data    - Initial data from which to construct the Roll Table
+   * @param context - Construction context options
+   */
+  constructor(data: BaseRollTable.ConstructorData, context?: DocumentConstructionContext);
+
+  static override metadata: Readonly<BaseRollTable.Metadata>;
+
+  static override defineSchema(): BaseRollTable.Schema;
+
+  /**
+   * The default icon used for newly created Macro documents
+   */
+  static DEFAULT_ICON: "icons/svg/d20-grey.svg";
+
+  static override migrateData(source: object): object;
+
+  static override shimData(
+    data: object,
+    {
+      embedded
+    }?: {
+      /**
+       * Apply shims to embedded models?
+       * @defaultValue `true`
+       */
+      embedded?: boolean;
+    }
+  ): object;
+}
+export default BaseRollTable;
+
+declare namespace BaseRollTable {
+  type Metadata = Merge<
+    DocumentMetadata,
+    {
+      name: "RollTable";
+      collection: "tables";
+      indexed: true;
+      compendiumIndexFields: ["_id", "name", "img", "sort"];
+      embedded: { TableResult: "results" };
+      label: "DOCUMENT.RollTable";
+      labelPlural: "DOCUMENT.RollTables";
+    }
+  >;
+
+  type SchemaField = fields.SchemaField<Schema>;
+  type ConstructorData = UpdateData;
```

Comment:
This also needs to be updated, as in the BaseMacro PR.

Reply:
Updated.